src/libstd/sys_common/backtrace.rs

```diff
@@ -122,7 +158,7 @@ pub fn log_enabled() -> Option<PrintFormat> {
     }
 
     let val = match env::var_os("RUST_BACKTRACE") {
-        Some(x) => if &x == "0" {
+        Some(x) => if &x == "0" || &x == "" {
```

Comment:
Perhaps we could hold off on changing the interpretation of `RUST_BACKTRACE` for now? I'm wary to add more and more cases for positive/negative intepretation.